pkg/validator/noderesourcetopologies.go

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2022 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package validator
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	topologyclientset "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/clientset/versioned"
+
+	deployervalidator "github.com/k8stopologyawareschedwg/deployer/pkg/validator"
+)
+
+func CollectNodeResourceTopologies(ctx context.Context, _ client.Client, data *ValidatorData) error {
+
+	cli, err := getTopologyClient()
+	if err != nil {
+		return err
+	}
+
+	nrtList, err := cli.TopologyV1alpha1().NodeResourceTopologies().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		if IsNodeResourceTopologyCRDMissing(err) {
+			data.nrtCrdMissing = true
+			return nil
+		}
+		return err
+	}
+
+	data.nrtList = nrtList
+	return nil
+}
+
+func ValidateNodeResourceTopologies(data ValidatorData) ([]deployervalidator.ValidationResult, error) {
```

Comment:
the function returns the result within the if conditions, if we wish to return list of results, wouldn't it make sense to gather append all relevant results and then return the final results list, otherwise we don't need to return []deployervalidator.ValidationResult nor to append to ret, wdyt?

Reply:
also we don't have any case the func sets err, why to return error?

Reply:
I see your point here. The problem is that we are constraint by the functio signature - because validations want to be generic and return a list of validation errors, while in this specific case all the validations we implemented are mutually exclusive. So we will always either have zero or one validation errors, which we need to wrap in a slice anyway.
I don't see easy fixes without overhauling the API. Do you perhaps see a better approach here to solve this dilemma?

Reply:
> also we don't have any case the func sets err, why to return error?

another case on which we are constrained by a more generic API, which doesn't make much sense for this specific set of validation implemented here. :\

Reply:
I don't have ideas for this, but anyway it is no big deal to keep it the way it is